compiler/src/dotty/tools/dotc/config/CliCommand.scala

```diff
@@ -191,3 +144,56 @@ trait CliCommand:
 
   extension [T](setting: Setting[T])
     protected def value(using ss: SettingsState): T = setting.valueIn(ss)
+
+  extension (s: String)
+    def padLeft(width: Int): String = StringBuilder().tap(_.append(" " * (width - s.length)).append(s)).toString
```

Comment:
```suggestion
    def padLeft(width: Int): String = String.format("%1$" + width "s", s)
```

Reply:
I have been preferring "just emit the string" to "parse a format template then construct a string" because people complain about slow strings. Not sure if I would push back here because format is easier to read.

Reply:
> format is easier to read.
This. Code is read more often than writ. 

We need ability to create strings lazily and efficiently. Also `f"%${width}s"` could actually detect that the interpolated value is part of the format and translate to padding, etc.

Reply:
Otherwise we could simply have 
```diff
def padLeft(width: Int): String = StringBuilder().tap(_.append(" " * (width - s.length)).append(s)).toString
def padLeft(width: Int): String = " " * (width - s.length) + s
```
There is a single `String.+`, the builder will not help for this use case.